examples/fabric-bridge-app/fabric-bridge-common/src/BridgedAdministratorCommissioning.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "BridgedAdministratorCommissioning.h"
+
+#include "BridgedDevice.h"
+#include "BridgedDeviceManager.h"
+#include <app/AttributeAccessInterfaceRegistry.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::AdministratorCommissioning;
+
+CHIP_ERROR BridgedAdministratorCommissioning::Init()
+{
+    // We expect to be initialized after all the embr pluggin cluster initialization have been called,
+    // including the AdministratorCommissioning. This expectation allows us to get and unregister the
+    // existing AccessAttributeInterface for AdministratorCommissioning and register ourselves so that
+    // we can first see if the ReadAttribute is for us.
+
+    // We expect initialization after all embr plugin clusters initialization. This allows us to unregister
+    // the existing AccessAttributeInterface for AdministratorCommissioning and register ourselves, ensuring
+    // we get the callback for reading attribute. If the read is not intended for a bridged device we will
+    // forward it to the original attribute interface that we are unregistering.
+    mOriginalAttributeInterface = AttributeAccessInterfaceRegistry::Instance().Get(0, AdministratorCommissioning::Id);
+    VerifyOrReturnError(mOriginalAttributeInterface, CHIP_ERROR_INTERNAL);
+    AttributeAccessInterfaceRegistry::Instance().Unregister(mOriginalAttributeInterface);
+    VerifyOrDie(AttributeAccessInterfaceRegistry::Instance().Register(this));
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR BridgedAdministratorCommissioning::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    VerifyOrDie(aPath.mClusterId == Clusters::AdministratorCommissioning::Id);
+    EndpointId endpointId  = aPath.mEndpointId;
+    BridgedDevice * device = BridgeDeviceMgr().GetDevice(endpointId);
+
+    if (!device)
+    {
```

Comment:
if the device is not available on this endpoint, we just need to pass through to go to the default cluster implementation, right? why we need to call mOriginalAttributeInterface->Read(aPath, aEncoder);

Reply:
mOriginalAttributeInterface is the original default cluster implementation

Reply:
Actually, we don't need to use mOriginalAttributeInterface to access the default implementation. From the AttributeAccessInterface comment, we just need to return success without attempting to encode any data using aEncoder

Return success and not attempt to encode any data using aEncoder.  In
     *    this case, Ember attribute access will happen for the read. This may
     *    involve reading from the attribute store or external attribute
     *    callbacks.

Reply:
embr attribute access doesn't work for this cluster

https://github.com/project-chip/connectedhomeip/blob/master/src/app/clusters/administrator-commissioning-server/administrator-commissioning-server.cpp

This cluster relies on AttributeAccessInterface and not on embr to make sure the attributes are properly read back to client

Reply:
If we return without touching aEncoder, it should get to the default implementation at https://github.com/project-chip/connectedhomeip/blob/master/src/app/clusters/administrator-commissioning-server/administrator-commissioning-server.cpp

Lets land this as is, I will check why it does not work as the interface described when I got a chance later